docs/managing_deis/running-stateless-cluster.rst

```diff
@@ -0,0 +1,214 @@
+dependency of the Deis control plane.
+
+.. note::
+
+  This guide was adapted from content graciously provided by Deis community member
+  `Arne-Christian Blystad`_.
+
+Requirements
+------------
+
+External services are required to replace the internal store components:
+
+* S3-compatible blob store (like `Amazon S3`_)
+* PostgreSQL database (like `Amazon RDS`_)
+* Log drain service with syslog log format compatibility (like `Papertrail`_)
+
+Understanding component changes
+-------------------------------
+
+Either directly or indirectly, all components in the :ref:`control-plane`
+require Ceph (:ref:`store`). Some components require changes to accommodate
+the removal of Ceph. The necessary changes are described below.
+
+Logger
+^^^^^^
+
+The :ref:`logger` component provides a syslog-compatible endpoint to consume
+application logs, which it writes to a shared Ceph filesystem. These logs are
+read by the :ref:`controller` component. The :ref:`logspout` talks to the Docker
+daemon on each host, listens for log events from running applications, and ships
+them to the logger.
+
+The Logger component is not necessary in a Ceph-less Deis cluster. Instead of
+using the Logger, we will route all the logs directly to another syslog
+compatible endpoint.
+
+Database
+^^^^^^^^
+
+The :ref:`database` runs PostgreSQL and uses the Ceph S3 API (provided by
+``deis-store-gateway``) to store PostgreSQL backups and WAL logs.
+Should the host running database fail, the database component will fail over to
+a new host, start up, and replay backups and WAL logs to recover to its
+previous state.
+
+We will not be using the database component in the Ceph-less cluster, and will
+instead rely on an external database.
+
+Controller
+^^^^^^^^^^
+
+The :ref:`controller` component hosts the API that the Deis CLI consumes. The controller
+mounts the same Ceph filesystem that the logger writes to. When users run ``deis logs``
+to view an application's log files, the controller reads from this shared filesystem.
+
+A Ceph-less cluster will not store logs (instead sending them to an external service),
+so the ``deis logs`` command will not work for users.
+
+Registry
+^^^^^^^^
+
+The :ref:`registry` component is an instance of the offical Docker registry, and
+is used to store application releases. The registry supports any S3 store, so
+a Ceph-less cluster will simply reconfigure registry to use another store (typically
+Amazon S3 itself).
+
+Builder
+^^^^^^^
+
+The :ref:`builder` component is responsible for building applications deployed
+to Deis via the ``git push`` workflow. It pushes to registry to store releases,
+so it will require no changes.
+
+Store
+^^^^^
+
+The :ref:`store` components implement Ceph itself. In a Ceph-less cluster, we
+will skip the installation and starting of these components. We will also
+mock a fake ``deis-store-volume`` component so that the controller and logger
```

Comment:
Previously we write that we will not be running logger, yet here we say that the mock deis-store-volume is required for the logger. I suggest removing the logger part of the line :)